app/workers/deletes.php

```diff
@@ -245,25 +249,51 @@ protected function deleteMemberships(Document $document, string $projectId): voi
 
     /**
      * @param Document $document project document
+     * @throws Exception
      */
     protected function deleteProject(Document $document): void
     {
         $projectId = $document->getId();
 
-        // Delete all DBs
-        $this->getProjectDB($projectId)->delete($projectId);
+        // Delete project tables
+        $dbForProject = $this->getProjectDB($projectId, $document);
+
+        while (true) {
+            $collections = $dbForProject->listCollections();
+
+            if (empty($collections)) {
+                break;
+            }
+
+            foreach ($collections as $collection) {
+                $dbForProject->deleteCollection($collection->getId());
+            }
+        }
+
+        // Delete metadata tables
+        try {
```

Comment:
I don't see where we delete the project document itself?

Reply:
Thats done in the controller itself 
https://github.com/appwrite/appwrite/blob/master/app/controllers/api/projects.php#L642